consul/src/main/java/com/linecorp/armeria/client/consul/ConsulEndpointGroup.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.consul;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.Nullable;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linecorp.armeria.client.ClientRequestContextCaptor;
+import com.linecorp.armeria.client.Clients;
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.client.endpoint.DynamicEndpointGroup;
+import com.linecorp.armeria.client.endpoint.EndpointGroup;
+import com.linecorp.armeria.internal.consul.ConsulClient;
+
+import io.netty.channel.EventLoop;
+
+/**
+ * A Consul-based {@link EndpointGroup} implementation that retrieves the list of {@link Endpoint}s
+ * from Consul using <a href="https://www.consul.io/api">Consul's RESTful HTTP API</a> and
```

Comment:
RESTful and HTTP mean the same thing in this context so I think we can just use the word in the web site that is
`Consul's HTTP API`

Reply:
Ok.

Reply:
Ok.

Reply:
Gentle ping 😄